### ISSUE TITLE
Influence of batch size of training convergence

### ISSUE THREAD

**Issue by CDitzel** [lucidrains/DALLE-pytorch#31]
I asked the same question in the[ official CLIP repository](https://github.com/openai/CLIP/issues/25).

According to the paper they use a large batch size of ~32k samples which means that the raw untrained network initially has a chance of ~1/32k of predicting the correct pair. 

I am wondering, how the convergence/learning process would differ, if instead a binary classification problem was formulated and the network would be presented with matching text/image pairs and non-matching pairs alternatingly and be tasked with predicting whether those samples are actually in agreement or not. 

In other words, what does the softmax over ~32k entries prior to cross entropy calculation bring to the table which cannot be achieved more conveniently by using sigmoid and binary cross entropy to predict matching/non-matching pairs. As a side effect this would also abolish the dependence on the batch size which seems to be rather crucial?
